samples/nrf9160/lte_ble_gateway/src/main.c

```diff
@@ -146,34 +150,85 @@ void nrf_modem_recoverable_error_handler(uint32_t err)
 	error_handler(ERROR_MODEM_RECOVERABLE, (int)err);
 }
 
-/**@brief Callback for GPS events */
-static void gps_handler(const struct device *dev, struct gps_event *evt)
+/**@brief Request assisted GPS data from nRF Cloud. */
+static void agps_request(struct k_work *work)
+{
+	int retval;
+	struct nrf_modem_gnss_agps_data_frame agps_data;
+
+	if (!nrf_cloud_agps_request_in_progress()) {
+		retval = nrf_modem_gnss_read(&agps_data, sizeof(agps_data),
+					     NRF_MODEM_GNSS_DATA_AGPS_REQ);
+		if (retval) {
+			LOG_ERR("Failed to read AGPS data, err %d", retval);
+			return;
+		}
+
+		retval = nrf_cloud_agps_request(&agps_data);
+		if (retval) {
+			LOG_ERR("Failed to request AGPS data, err %d", retval);
+			return;
+		}
+	}
+}
+
+/**@brief Callback for GNSS events */
+static void gnss_event_handler(int event)
 {
+	int retval;
+
 	uint32_t button_state, has_changed;
 	struct sensor_data in_data = {
-		.type = GPS_POSITION,
-		.length = evt->nmea.len,
+		.type = GNSS_POSITION,
 	};
 
-	ARG_UNUSED(dev);
+	switch (event) {
+	case NRF_MODEM_GNSS_EVT_PVT:
+		retval = nrf_modem_gnss_read(&last_pvt, sizeof(last_pvt), NRF_MODEM_GNSS_DATA_PVT);
```

Comment:
I don't think this should ever fail, but maybe it would still be good to check `retval`.